### ISSUE TITLE
Incorrect hosted zone match

### ISSUE THREAD

**Issue by sridhar562345** [zappa/Zappa#1190]
https://github.com/zappa/Zappa/blob/9b2a98ccaebe271e614c0af1bcf301ea6facb765/zappa/core.py#L3197-L3208

Suppose there are two hosted zones `api.in` and `beta.api.in` and domain name specified is `some-beta.api.in`

The current logic selects hosted `beta.api.in` as the hosted zone for the specified domain.

**Comment by monkut** [comment]
What is the expected result?

**Comment by sridhar562345** [comment]
> What is the expected result?

It should select `api.in` as the best match zone

**Comment by monkut** [comment]
@sridhar562345 I attempted an improvement PR, can you check it out and let me know if it meets your expectations?

**Comment by sridhar562345** [comment]
@monkut

I was actually working on it as well and I guess I have a better logic to verify this.

Here is my logic
```py
@staticmethod
def get_best_match_zone(all_zones, domain):
    """Return zone id which name is closer matched with domain name."""

    # Related: https://github.com/Miserlou/Zappa/issues/459
    public_zones = [zone for zone in all_zones['HostedZones'] if not zone['Config']['PrivateZone']]

    zones = {zone['Name'][:-1]: zone['Id'] for zone in public_zones if zone['Name'][:-1] == domain}

    if not zones:
        zones = {zone['Name'][:-1]: zone['Id'] for zone in public_zones if '.' + zone['Name'][:-1] in domain}

    if zones:
        keys = max(zones.keys(), key=lambda a: len(a))  # get longest key -- best match.
        return zones[keys]
    else:
        return None
```

**Comment by monkut** [comment]
Thanks!

**Comment by monkut** [comment]
released in 0.56.0, closing.
